src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/ModuleVelocity.kt

```diff
@@ -121,6 +125,50 @@ object ModuleVelocity : Module("Velocity", Category.COMBAT) {
 
     }
 
+    /**
+     *
+     * Basic velocity targeting specifically Hypixel.
+     */
+    private object Hypixel : Choice("Hypixel") {
+
+        override val parent: ChoiceConfigurable
+            get() = modes
+
+        val packetHandler = handler<PacketEvent> { event ->
+            val packet = event.packet
+
+            // Check if this is a regular velocity update
+            if (packet is EntityVelocityUpdateS2CPacket && packet.id == player.id) {
+                if (packet.velocityX == 0 && packet.velocityZ == 0) {
+                    // positive vertical velocity is a staff velocity check
+                    if (packet.velocityY > 0) {
+                        // alert the user
+                        notification(
+                            "Staff Detected",
+                            "Staff are watching you.",
+                            NotificationEvent.Severity.INFO
+                        )
+                        return@handler
+                    }
+                    // you can just cancel negative vertical velocity, it's not checked
+                    event.cancelEvent()
+                    return@handler
+                }
+
+                val currentVelocity = mc.player?.velocity ?: return@handler
```

Comment:
You can always access the player's velocity by typing `player.velocity` and do not need to do any null checks, because handler<PacketEvent> only runs in an inGame environment.